File: back-end/APIColegio/APIColegio/Models/Alumno.cs
using System.ComponentModel.DataAnnotations;

namespace APIColegio.Models
{
    public class Alumno
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string NombreAlumno { get; set; } = string.Empty;

        [Required]
        public DateTime FechaNacimiento { get; set; }

        [Required]
        [StringLength(100)]
        public string NombrePadre { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string NombreMadre { get; set; } = string.Empty;

        [Required]
        [StringLength(20)]
        public string Grado { get; set; } = string.Empty;

        [Required]
        [StringLength(10)]
        public string Seccion { get; set; } = string.Empty;

        [Required]
        public DateTime FechaIngreso { get; set; }
    }
}

File: front-end/src/types/alumno.ts
export interface Alumno {
  id: number;
  nombreAlumno: string;
  fechaNacimiento: string;
  nombrePadre: string;
  nombreMadre: string;
  grado: string;
  seccion: string;
  fechaIngreso: string;
  edad: number;
  gradoCompleto: string;
}

export interface CreateAlumnoDto {
  nombreAlumno: string;
  fechaNacimiento: string;
  nombrePadre: string;
  nombreMadre: string;
  grado: string;
  seccion: string;
  fechaIngreso: string;
}

export interface ApiResponse<T> {
  success: boolean;
  data: T;
  message: string;
  count: number;
  timestamp: string;
}
